x/ecocredit/params.go

```diff
@@ -37,12 +41,38 @@ func validateCreditClassFee(i interface{}) error {
 	return nil
 }
 
-func NewParams(creditClassFee sdk.Coins) Params {
+func validateAllowlistCreditCreators(i interface{}) error {
+	v, ok := i.([]string)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+	for _, sAddr := range v {
+		_, err := sdk.AccAddressFromBech32(sAddr)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateAllowlistEnabled(i interface{}) error {
+	_, ok := i.(bool)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	return nil
+}
+
+func NewParams(creditClassFee sdk.Coins, allowlist []string, allowlistEnabled bool) Params {
 	return Params{
-		CreditClassFee: creditClassFee,
+		CreditClassFee:               creditClassFee,
+		AllowedClassCreatorAddresses: allowlist,
+		AllowlistEnabled:             allowlistEnabled,
 	}
 }
 
+// TODO(tyler): what to put for default addresses???
```

Comment:
Yeah, I'm still not quite clear whether this default value will be used or if we'll set it manually in some new genesis config. @clevinson do you know what will happen here?

Reply:
yeah, thats the one thing im not too sure on - is enabled + empty allowlist a sane default here? @aaronc @cpolitano

Reply:
I think we should set disabled as default. The primary use case of defaults is in local testing (when someone runs `regen init`) and integration tests.

For mainnet (which is where we care about it being permissioned), we can set this explicitly to be enabled in the upgrade handler.